### ISSUE TITLE
[Bug] Adapter cannot be applied directly as described on paper

### ISSUE THREAD

**Issue by huchenlei** [showlab/X-Adapter#25]
In the paper, the author showed that the X-Adapter can be used to directly applied on SDXL checkpoint to denoise a pure latent noise and apply the plugable control condition (ControlNet, LoRA, etc). 
![image](https://github.com/showlab/X-Adapter/assets/20929282/b98dfab6-3a37-4d09-9fa9-6f3a0a357493)

However, according to my testing, this is not true. If I remove the sd15 pass, the SDXL model with SD15 ControlNet applied via adapter cannot produce img follow the control constraint.

## Steps to reproduce
- Install comfyui node: https://github.com/kijai/ComfyUI-Diffusers-X-Adapter
- Find `pipeline/pipeline_sd_xl_adapter_controlnet.py`
- Comment out following code: https://github.com/kijai/ComfyUI-Diffusers-X-Adapter/blob/81b77e441c2dd5566acf5025fa9cadfb8a574490/pipeline/pipeline_sd_xl_adapter_controlnet.py#L993-L1078
- Run the node
- Observe result not following canny edge control

![image](https://github.com/showlab/X-Adapter/assets/20929282/bf0abdd2-ff22-480a-b7e1-d6a6963da11f)

## Is this paper a scam?
I am suspecting if the xadapter is really doing its job on transfering the control of sd15 controlnet model to sdxl. If we are starting at T0 with the sd15 controlnet already applied with some steps, as long as the SDXL model not doing too much extra modification, the result should not deviate a lot from the T0 result.

Here I compare the result between 2 scenarios:
### No Adapter 
Add `up_block_additional_residual = None` after https://github.com/kijai/ComfyUI-Diffusers-X-Adapter/blob/81b77e441c2dd5566acf5025fa9cadfb8a574490/pipeline/pipeline_sd_xl_adapter_controlnet.py#L1166

Essentially this is doing generation with ControlNet using SD15 model, and then refine with SDXL model.
![image](https://github.com/showlab/X-Adapter/assets/20929282/1a89cb74-2841-4054-8369-a4b5b4e3447b)
![image](https://github.com/showlab/X-Adapter/assets/20929282/b1db0548-5ff9-44e5-ae7b-aef9358c129b)

### With Adapter
![image](https://github.com/showlab/X-Adapter/assets/20929282/27e98be3-a169-452d-bbf2-87971911ea8f)
![image](https://github.com/showlab/X-Adapter/assets/20929282/b5e063ae-3ebd-40d5-8717-258e011328ee)

I do not observe clear improvement on the X-Adapter.

**Comment by Semolce9** [comment]
Hi. I test canny controlnet using following bash scripts:
```
python inference.py --plugin_type "controlnet" \
--prompt "A cute cat, high quality, extremely detailed" \
--condition_type "canny" \
--input_image_path "./assets/CuteCat.jpeg" \
--controlnet_condition_scale_list 1.5 \
--adapter_guidance_start_list 1.00 \
--adapter_condition_scale_list 1.00 \
```
I set `adapter_guidance_start` to 1.00 which means only SDXL denoises a pure noise under X-Adapter's guidance. Given the condition:
![A cute cat_canny_condition](https://github.com/showlab/X-Adapter/assets/152716091/0d65ef8f-ecf7-4065-b15a-ef3a107cdcd3)
I get the result:
![A cute cat_0_ccs_1 50_ags_1 00_acs_1 00](https://github.com/showlab/X-Adapter/assets/152716091/9205758c-01b7-4387-8f42-a9d8df09c6c3)
This result is consistent with our ablation study (fig.10). We also shows a suboptimal result in fig.10 and it indicates that we need both initial latents from SD1.5 and guidance from X-Adapter.

**Comment by huchenlei** [comment]
Also, I would like to see explanation on why the result's quality does not change much with/without adapter.

**Comment by Semolce9** [comment]
Yes and I will add noise to SD1.5's output to timestep 50 which is pure noise

**Comment by huchenlei** [comment]
Can you confirm the result if you completely remove sd15 pass code and use initial latent directly? 

Comment out following code: https://github.com/kijai/ComfyUI-Diffusers-X-Adapter/blob/81b77e441c2dd5566acf5025fa9cadfb8a574490/pipeline/pipeline_sd_xl_adapter_controlnet.py#L993-L1078

**Comment by Semolce9** [comment]
It's improper to directly remove these code. You run `prepare_xl_latents_from_sd_1_5` and `sd1_5_add_noise` without generating latents_sd1_5_prior. During training, sd1.5 and sdxl's latent are naturally aligned together so x-adapter can only work on situations where two latents are aligned together even the alignment is weak(`adapter_guidance_start=1.00`).

I set `up_block_additional_residual` = None, `adapter_guidance_start` = 1.00 and given condition:
![A cute cat_canny_condition](https://github.com/showlab/X-Adapter/assets/152716091/d8fcac7d-6487-4343-b876-84d4fc91d638)
result:
![A cute cat_0_ccs_1 50_ags_1 00_acs_1 00](https://github.com/showlab/X-Adapter/assets/152716091/448d77d7-c1f4-4876-a88c-6947f5beb9de)
Using x-adapter, setting `controlnet_condition_scale`=1.75, `adapter_guidance_start`=1.00, `adapter_condition_scale`=1.00, I get following result:
![A cute cat_0_ccs_1 75_ags_1 00_acs_1 00](https://github.com/showlab/X-Adapter/assets/152716091/090df610-bb7e-446b-a58d-c31852c69a28)

If your convern lies in "why do the author mention we can directly using it?". The answer is that we want to make comparison between two methods in our paper and the result shows that we do not recommand users to directly apply x-adapter to sdxl since this is not what x-adapter is trained on.

**Comment by huchenlei** [comment]
I think if the full sd15 pass is mandatory, the figure used in the paper is misleading. The figure is showing that the adapter can be directly applied on a random noise.

**Comment by Semolce9** [comment]
I agreed. Thank you for suggestion. I will clarify it and upload a new version to arxiv.

**Comment by huchenlei** [comment]
I have intermidiate results dumped in ComfyUI. It seems like either the ComfyUI result is broken or the impl of ControlNet pipeline is broken? https://github.com/kijai/ComfyUI-Diffusers-X-Adapter/pull/15

adapter_guidance_start = 1.0
![image](https://github.com/showlab/X-Adapter/assets/20929282/c36b07c4-d031-423d-8823-20278622ec83)

adapter_guidance_start = 0.5
![image](https://github.com/showlab/X-Adapter/assets/20929282/c7da5959-96ec-4a9a-a9a2-8575595739bd)

Please confirm that these result can be reproduced. Also, is there any requirement on what checkpoints to use? (SD15/SDXL)

**Comment by Semolce9** [comment]
If images from left to right, top to bottom is sd1.5's output, sdxl's output, sd1.5's input, sdxl's input, then the result is as expected. We can see better constraint with lower `adapter_guidance_start`. sd1.5 and sdxl's input is noise if you decode it to pixel space. If you want better result, please try different values of `controlnet_condition_scale` and `adapter_condition_scale`.

I cannot check all checkpoints since there are so many. I did experiments using official sd1.5 and sdxl developed by stability.ai. I saw some good cases using other checkpoints like https://twitter.com/ZHOZHO672070/status/1759516726125908361.

**Comment by light-and-ray** [comment]
I think it's enough to know that it requires full sd1.5 checkpoint. If it requires, it cannot be used just as "adapter", like loras, controlnets, ip-adapters, lcm loras etc :(

**Comment by LastTargaryen** [comment]
@light-and-ray, I have a strong feeling that this adapter is...unusable. It's a failed experiment. Should not have been released the way it has been. It's very misleading and gives the impression of being beneficial to the ML community and the SD-user community--but it's not. Not even for advanced users.
